typed-racket-lib/typed-racket/base-env/base-env-numeric.rkt

```diff
@@ -781,27 +782,27 @@
   (-> N B : (-PS (-is-type 0 (Un -RealZeroNoNan -InexactComplex -InexactImaginary))
                  (-not-type 0 -RealZeroNoNan)))]
 
-[number? (make-pred-ty N)]
+[number? (make-pred-ty N #t)]
```

Comment:
Hang on, I think we do want edits to appear on all these lines. The `#t` is telling Shallow TR it can trust the result of `number?`. If that's wrong, we'd have a soundness bug.

How about a rename to make this more clear: `make-unsafe-pred-ty`? `make-unsafe-shallow-pred-ty` ?

Reply:
`make-unsafe-shallow-pred-ty` looks good to me